src/Integration.Vsix/Analysis/RuleSettingsMapper.cs

```diff
@@ -0,0 +1,44 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2024 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System.ComponentModel.Composition;
+using SonarLint.VisualStudio.Core;
+using SonarLint.VisualStudio.SLCore.Service.Rules.Models;
+
+namespace SonarLint.VisualStudio.Integration.Vsix.Analysis;
+
+public interface IRuleSettingsMapper
+{
+    Dictionary<string, StandaloneRuleConfigDto> MapRuleSettingsToSlCoreSettings(RulesSettings rulesSettings);
+}
+
+[Export(typeof(IRuleSettingsMapper))]
+[PartCreationPolicy(CreationPolicy.Shared)]
+public class RuleSettingsMapper : IRuleSettingsMapper
+{
+    public Dictionary<string, StandaloneRuleConfigDto> MapRuleSettingsToSlCoreSettings(RulesSettings rulesSettings)
```

Comment:
Would it make sense to move the SLCoreServiceProvider dependency from the UserSettingsProvider to this class instead? So the slcore integration is fully contained within this class?

Reply:
I agree. That task would require quite some refactoring so I would rather leave it as it is for now.